tests/test_brownian_interval.py

```diff
@@ -0,0 +1,311 @@
+
+
+def _levy_returns():
+    yield "none", False, False
+    yield "space-time", False, False
+    yield "space-time", True, False
+    for levy_area_approximation in ('davie', 'foster'):
+        for return_U in (True, False):
+            for return_A in (True, False):
+                yield levy_area_approximation, return_U, return_A
+
+
+@pytest.mark.parametrize("device", devices)
+@pytest.mark.parametrize("levy_area_approximation, return_U, return_A", _levy_returns())
+def test_shape(device, levy_area_approximation, return_U, return_A):
+    if device == gpu and not torch.cuda.is_available():
+        pytest.skip(msg="CUDA not available.")
+
+    for shape, A_shape in (((SMALL_BATCH_SIZE, D), (SMALL_BATCH_SIZE, D, D)),
+                           ((SMALL_BATCH_SIZE,), (SMALL_BATCH_SIZE,)),
+                           ((), ())):
+        ta, tb, bm = _setup(device, levy_area_approximation, shape)
+        sample1 = bm(ta, return_U=return_U, return_A=return_A)
+        sample2 = bm(tb, return_U=return_U, return_A=return_A)
+        sample3 = bm(ta, tb, return_U=return_U, return_A=return_A)
+        shapes = []
+        A_shapes = []
+        for sample in (sample1, sample2, sample3):
+            if return_U:
+                if return_A:
+                    W1, U1, A1 = sample
+                    shapes.append(W1.shape)
+                    shapes.append(U1.shape)
+                    A_shapes.append(A1.shape)
+                else:
+                    W1, U1 = sample
+                    shapes.append(W1.shape)
+                    shapes.append(U1.shape)
+            else:
+                if return_A:
+                    W1, A1 = sample
+                    shapes.append(W1.shape)
+                    A_shapes.append(A1.shape)
+                else:
+                    W1 = sample
+                    shapes.append(W1.shape)
+
+        for shape_ in shapes:
+            assert shape_ == shape
+        for shape_ in A_shapes:
+            assert shape_ == A_shape
+
+
+@pytest.mark.parametrize("device", devices)
+@pytest.mark.parametrize("levy_area_approximation, return_U, return_A", _levy_returns())
+def test_determinism_simple(device, levy_area_approximation, return_U, return_A):
+    if device == gpu and not torch.cuda.is_available():
+        pytest.skip(msg="CUDA not available.")
+
+    ta, tb, bm = _setup(device, levy_area_approximation, (SMALL_BATCH_SIZE, D))
+    vals = [bm(ta, tb, return_U=return_U, return_A=return_A) for _ in range(REPS)]
+    for val in vals[1:]:
+        if torch.is_tensor(val):
+            val = (val,)
+        if torch.is_tensor(vals[0]):
+            val0 = (vals[0],)
+        else:
+            val0 = vals[0]
+        for v, v0 in zip(val, val0):
+            assert (v == v0).all()
+
+
+@pytest.mark.parametrize("device", devices)
+@pytest.mark.parametrize("levy_area_approximation, return_U, return_A", _levy_returns())
+def test_determinism_large(device, levy_area_approximation, return_U, return_A):
```

Comment:
Could you add a short descriptive docstring for this test, since the function body is a little longer than the others.

Reply:
Sure.